Code/Framework/AzFramework/Platform/Android/AzFramework/Windowing/NativeWindow_Android.cpp

```diff
@@ -39,22 +42,22 @@ namespace AzFramework
 
         int windowWidth = 0;
         int windowHeight = 0;
+        // Window size is determinate by the OS. We cannot set it like in other systems.
         if (AZ::Android::Utils::GetWindowSize(windowWidth, windowHeight))
         {
-            // Use native window size from the device if available
+            // Use the size returned by the OS (most likely the fullscreen size).
             m_width = static_cast<uint32_t>(windowWidth);
             m_height = static_cast<uint32_t>(windowHeight);
         }
         else
         {
+            AZ_Error("NativeWindow", false, "Failed to get native window size");
             m_width = geometry.m_width;
             m_height = geometry.m_height;
         }
 
-        if (m_nativeWindow)
-        {
-            ANativeWindow_setBuffersGeometry(m_nativeWindow, m_width, m_height, ANativeWindow_getFormat(m_nativeWindow));
-        }
+        // Set the resolution to the same size of the window.
+        ANativeWindow_setBuffersGeometry(m_nativeWindow, m_width, m_height, ANativeWindow_getFormat(m_nativeWindow));
```

Comment:
Is there a reason the nullptr check was removed?

Reply:
Probably because `m_nativeWindow` should not be nullptr by design. `m_nativeWindow` is set on line 41, at the very least, an AZ_Assert to check if its nullptr should be placed there.